library/core/src/cell.rs

```diff
@@ -1487,7 +1487,7 @@ impl<'b, T: ?Sized + Unsize<U>, U: ?Sized> CoerceUnsized<Ref<'b, U>> for Ref<'b,
 #[stable(feature = "std_guard_impls", since = "1.20.0")]
 impl<T: ?Sized + fmt::Display> fmt::Display for Ref<'_, T> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.value.fmt(f)
+        (**self).fmt(f)
```

Comment:
ymmv: `<T as fmt::Display>::fmt(self, f)` would be another way to ensure that this bug can't really get reintroduced.

As you have it is fine too.

Reply:
I thought about that, but it should be fine with the added test coverage.